### PR TITLE
Record array bug

### LINKED ISSUE #767: The `matchingMode` field missing after editing FTP service configurations
### Description

When editing the configuration of an FTP service, the `matchingMode` field inside `fileDependencyConditions` is lost. This results in an incomplete configuration and the field is no longer visible in the UI.

- When adding the configurations for the first time, the `matchingMode` field is correctly set and visible in the UI.
Image
</br></br>

- However, after saving and attempting to update the configurations again, the `matchingMode` value is missing from the configuration. Also the field is no longer visible in the UI.
Image

### Steps to Reproduce

1. Configure an FTP service with the following configuration in WSO2 Integrator.

```json
{
    "path": "/",
    "fileAgeFilter": {
        "minAge": 0.0d,
        "maxAge": 0.0d
    },
    "fileDependencyConditions": [
        {
            "targetPattern": "",
            "requiredFiles": [],
            "matchingMode": "ALL"
        }
    ]
}
```

2. Open the FTP service configuration in the UI.
3. Try to modify any field there.

Then you may observe the value for the matchingField is not visible in the configs.

### Version

5.0.0-alpha6

### Environment Details (with versions)

_No response_

### PR DESCRIPTION
**Pull request by niveathika**
Fixes https://github.com/wso2/product-integrator/issues/767

## Summary

This pull request fixes a bug where enum-typed fields inside arrays of record elements were lost when configurations were re-parsed and saved. The issue manifested as enum values (such as `matchingMode` in FTP service configurations) disappearing after editing and reopening configurations.

## Root Cause

The `EnumType` class lacked a `copy()` override method, causing enum member information to be dropped during deep-cloning operations when `RecordType.copy()` was invoked on types containing arrays of records.

## Changes

**EnumType.java:**
- Added a `copy()` override method that performs deep-cloning of the enum members list
- Ensures each enum member is properly cloned and all type metadata is preserved
- Propagates inherited fields through the cloning process

**EnumTypeCopyTest.java:**
- Added comprehensive test coverage for the `EnumType.copy()` functionality
- Validates that cloning creates distinct instances with deep-copied members
- Tests edge cases including null and empty member lists
- Verifies that enum member preservation propagates correctly through nested type structures (records containing enums, arrays containing records with enums)

## Outcome

Enum-typed fields within arrayed record elements are now properly preserved through configuration re-parse and save cycles, ensuring consistent data persistence and correct UI rendering.

### CODE REVIEW

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/types/EnumType.java [hunk 1]
```diff
@@ -43,4 +44,16 @@ public void setMembers(List<Type> members) {
...
+    @Override
+    public EnumType copy() {
+        List<Type> copiedMembers = new ArrayList<>();
+        if (this.members != null) {
+            for (Type member : this.members) {
+                copiedMembers.add(member.copy());
+            }
+        }
+        EnumType copy = new EnumType(copiedMembers);
+        copyBaseFields(copy);
```
[the remark this answers is not in the document]
Comment (niveathika) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/types/EnumType.java:56]: This is not possible LHS with null values.

**Comment by NipunaRanasinghe** [comment]
@niveathika Shall we update the LS unit test suite to cover the fixed scenario?

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/types/EnumType.java [hunk 2]
```diff
@@ -43,4 +44,16 @@ public void setMembers(List<Type> members) {
...
 
+    @Override
+    public EnumType copy() {
+        List<Type> copiedMembers = new ArrayList<>();
+        if (this.members != null) {
+            for (Type member : this.members) {
+                copiedMembers.add(member.copy());
+            }
+        }
+        EnumType copy = new EnumType(copiedMembers);
```
[the remark this answers is not in the document]
Comment (niveathika) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/types/EnumType.java:55]: N/A

**Comment by niveathika** [comment]
@NipunaRanasinghe Added in [39d0f36](https://github.com/ballerina-platform/ballerina-language-server/pull/880/commits/39d0f36d7ce97f98ded232002ec50e5c3a1cd8cc)

**Review by nipunayf — APPROVED** [review]
Similar to `enum`, the following candidates may also be affected and may need to be checked separately.

- ErrorType (line 26) with isErrorUnion, errorUnion, and detailType.
- TableType (line 29) with rowType, keys, and constraintType.
- ObjectType (line 29) with fields.
- MapType (line 27) with paramType.
- IntersectionType (line 29) with members.
- StreamType (line 28) with leftTypeParam and rightTypeParam.
- InclusionType (line 26) with inclusionType
